src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -757,8 +758,8 @@ void applyBoosting_(const UInt i,
 		    const vector<Real>& actualDensity,
 		    const Real boost,
 	            vector<Real>& output) {
-  if(boost < htm::Epsilon) return; //skip for disabled boosting
-  output[i] = exp((targetDensity - actualDensity[i]) * boost); //TODO doc this code
+  //output[i] = exp((targetDensity - actualDensity[i]) * boost); //exponential boosting, default for Numenta
+  output[i] = log(actualDensity[i]) / log(targetDensity);
```

Comment:
Performance optimizations, which helped when I did this in the `columnPooler2` branch.
- Move the `log(targetDensity)` to the outermost loop, since it is constant across all of the mini-columns.  In this case calling code contains the loop which you should avoid, not this function.
- Use `log2` instead of `log`

Reply:
>     log2 instead of log

done

> log(targetDensity) to the outermost loop, since it is constant

the problem here is I still need to keep local inh working, which changes the density for each column